Fix build failures in screenshot targets after adding ScreenObject

mokagio opened the pull request:
Sorry about missing them 🤦‍♂️ 😳 

To test, try building any of the two screenshots targets in `develop` and notice they'll fail. On this branch, they'll pass. Also notice CI is green, [including the UI tests](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-iOS/23619/workflows/ab888b8e-c2e9-4d5c-b000-ffdf412df287).

~~To avoid making this mistake again, I instructed the UI tests scheme to also build the screenshots targets. This should give us faster feedback on the need to update files from those targets in the future.~~ **Update** I had to remove this because, after the changes in how the code signing secrets are generated (#16806) building the Jetpack screenshots together with the WordPress ones resulted in two calls to the `Secrets.swift` generation and the build failed because multiple phases generated that same file.

Ideally, as we move more logic in the shared UI tests frameworks, this kind of errors will become less likely to occur because most things will be centralized.

## Regression Notes

1. Potential unintended areas of impact
N.A.


2. What I did to test those areas of impact (or what existing automated tests I relied on)
N.A.

3. What automated tests I added (or what prevented me from doing so)
N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

mokagio committed:
Link ScreenObject in JetpackScreenshotGeneration

mokagio committed:
Address Jetpack screenshots build failure due to unhandled throwing call

mokagio committed:
Address Jetpack screenshots build failure due to unhandled throwing call

mokagio committed:
Link ScreenObject in WordPressScreenshotGeneration target

mokagio committed:
Link ScreenObject to UITestsFoundation to simplify screenshots setup

This way, they both get it without having to explicitly link to it.

mokagio committed:
Remove missing "UITests" target from WordPressUITests scheme

mokagio committed:
Remove unused WordPressTest target from WordPressUITests scheme

pachlava reviewed (approved):
I built both `WordPressScreenshotGeneration` and `JetpackScreenshotGeneration` with no throw-related errors (and seen both have these errors in `develop`). Thanks, Gio!

mokagio committed:
Remove ScreenObject linking from WordPressUITests

WordPressUITests gets access to it already via UITestsFoundation.

mokagio committed:
Build WordPress screenshots target together with UI tests one

This is to ensure they keep building without having to manually cycle
through all both tests target, since WordPressUITests is the main one
and the one we run in CI.

mokagio commented:
Notice the update in the PR description:

> I had to remove this [building both Jetpack and WordPress screenshots with the UI tests] because, after the changes in how the code signing secrets are generated (#16806) building the Jetpack screenshots together with the WordPress ones resulted in two calls to the Secrets.swift generation and the build failed because multiple phases generated that same file.

I did that with an interactive rebase, to keep the history clean and each commit (?) building. I'm merging this now, after running the UI tests in CI and see them pass.
